# Importar Librerias

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Análisis breve de los distintos datasets

In [11]:
car_data = pd.read_csv('../datasets/car data.csv')
car_details = pd.read_csv('../datasets/CAR DETAILS FROM CAR DEKHO.csv')
car_details_v3  = pd.read_csv('../datasets/Car details v3.csv')
car_details_v4  = pd.read_csv('../datasets/Car details v4.csv')

In [20]:
car_data.sample(5)

Index(['Car_Name', 'Year', 'Selling_Price', 'Present_Price', 'Kms_Driven',
       'Fuel_Type', 'Seller_Type', 'Transmission', 'Owner'],
      dtype='object')

In [13]:
car_data.describe(include='all').T

count unique     top freq         mean           std     min  \
Car_Name         301     98    city   26          NaN           NaN     NaN   
Year           301.0    NaN     NaN  NaN  2013.627907      2.891554  2003.0   
Selling_Price  301.0    NaN     NaN  NaN     4.661296      5.082812     0.1   
Present_Price  301.0    NaN     NaN  NaN     7.628472      8.644115    0.32   
Kms_Driven     301.0    NaN     NaN  NaN  36947.20598  38886.883882   500.0   
Fuel_Type        301      3  Petrol  239          NaN           NaN     NaN   
Seller_Type      301      2  Dealer  195          NaN           NaN     NaN   
Transmission     301      2  Manual  261          NaN           NaN     NaN   
Owner          301.0    NaN     NaN  NaN     0.043189      0.247915     0.0   

                   25%      50%      75%       max  
Car_Name           NaN      NaN      NaN       NaN  
Year            2012.0   2014.0   2016.0    2018.0  
Selling_Price      0.9      3.6      6.0      35.0  
Present_Price      1.2      6.4      9.9      92.6  
Kms_Driven     15000.0  32000.0  48767.0  500000.0  
Fuel_Type          NaN      NaN      NaN       NaN  
Seller_Type        NaN      NaN      NaN       NaN  
Transmission       NaN      NaN      NaN       NaN  
Owner              0.0      0.0      0.0       3.0

In [21]:
car_details.sample(5)

Index(['name', 'year', 'selling_price', 'km_driven', 'fuel', 'seller_type',
       'transmission', 'owner'],
      dtype='object')

In [15]:
car_details.describe(include='all').T

count unique                     top  freq           mean  \
name             4340   1491  Maruti Swift Dzire VDI    69            NaN   
year           4340.0    NaN                     NaN   NaN    2013.090783   
selling_price  4340.0    NaN                     NaN   NaN  504127.311751   
km_driven      4340.0    NaN                     NaN   NaN   66215.777419   
fuel             4340      5                  Diesel  2153            NaN   
seller_type      4340      3              Individual  3244            NaN   
transmission     4340      2                  Manual  3892            NaN   
owner            4340      5             First Owner  2832            NaN   

                         std      min        25%       50%       75%  \
name                     NaN      NaN        NaN       NaN       NaN   
year                4.215344   1992.0     2011.0    2014.0    2016.0   
selling_price  578548.736139  20000.0  208749.75  350000.0  600000.0   
km_driven       46644.102194      1.0    35000.0   60000.0   90000.0   
fuel                     NaN      NaN        NaN       NaN       NaN   
seller_type              NaN      NaN        NaN       NaN       NaN   
transmission             NaN      NaN        NaN       NaN       NaN   
owner                    NaN      NaN        NaN       NaN       NaN   

                     max  
name                 NaN  
year              2020.0  
selling_price  8900000.0  
km_driven       806599.0  
fuel                 NaN  
seller_type          NaN  
transmission         NaN  
owner                NaN

In [16]:
car_details_v3.sample(5)

name  year  selling_price  km_driven  \
3522                      Maruti Wagon R LXI  2013         225000      58343   
5646                    Hyundai Santro Magna  2019         434999      19500   
1881                       Maruti Ciaz S 1.3  2018        1000000      40000   
2104                 Hyundai Elantra CRDi SX  2015         900000     145000   
3210  Tata Indica Vista Terra Quadrajet 1.3L  2010         105000     120000   

        fuel       seller_type transmission         owner     mileage  \
3522  Petrol  Trustmark Dealer       Manual   First Owner  21.79 kmpl   
5646  Petrol            Dealer       Manual   First Owner   20.3 kmpl   
1881  Diesel        Individual       Manual   First Owner  28.09 kmpl   
2104  Diesel        Individual       Manual   First Owner   22.7 kmpl   
3210  Diesel        Individual       Manual  Second Owner   18.0 kmpl   

       engine  max_power                  torque  seats  
3522   998 CC  67.05 bhp           90Nm@ 3500rpm    5.0  
5646  1086 CC  68.07 bhp        99.07Nm@ 4500rpm    5.0  
1881  1248 CC   88.5 bhp          200Nm@ 1750rpm    5.0  
2104  1582 CC  126.2 bhp  259.88Nm@ 1900-2750rpm    5.0  
3210  1248 CC     75 bhp          190Nm@ 1750rpm    5.0

In [17]:
car_details_v3.describe(include='all').T

count unique                     top  freq           mean  \
name             8128   2058  Maruti Swift Dzire VDI   129            NaN   
year           8128.0    NaN                     NaN   NaN    2013.804011   
selling_price  8128.0    NaN                     NaN   NaN  638271.807702   
km_driven      8128.0    NaN                     NaN   NaN   69819.510827   
fuel             8128      4                  Diesel  4402            NaN   
seller_type      8128      3              Individual  6766            NaN   
transmission     8128      2                  Manual  7078            NaN   
owner            8128      5             First Owner  5289            NaN   
mileage          7907    393               18.9 kmpl   225            NaN   
engine           7907    121                 1248 CC  1017            NaN   
max_power        7913    322                  74 bhp   377            NaN   
torque           7906    441          190Nm@ 2000rpm   530            NaN   
seats          7907.0    NaN                     NaN   NaN       5.416719   

                         std      min       25%       50%       75%  \
name                     NaN      NaN       NaN       NaN       NaN   
year                4.044249   1983.0    2011.0    2015.0    2017.0   
selling_price  806253.403508  29999.0  254999.0  450000.0  675000.0   
km_driven       56550.554958      1.0   35000.0   60000.0   98000.0   
fuel                     NaN      NaN       NaN       NaN       NaN   
seller_type              NaN      NaN       NaN       NaN       NaN   
transmission             NaN      NaN       NaN       NaN       NaN   
owner                    NaN      NaN       NaN       NaN       NaN   
mileage                  NaN      NaN       NaN       NaN       NaN   
engine                   NaN      NaN       NaN       NaN       NaN   
max_power                NaN      NaN       NaN       NaN       NaN   
torque                   NaN      NaN       NaN       NaN       NaN   
seats               0.959588      2.0       5.0       5.0       5.0   

                      max  
name                  NaN  
year               2020.0  
selling_price  10000000.0  
km_driven       2360457.0  
fuel                  NaN  
seller_type           NaN  
transmission          NaN  
owner                 NaN  
mileage               NaN  
engine                NaN  
max_power             NaN  
torque                NaN  
seats                14.0

In [22]:
car_details_v3.columns

Index(['name', 'year', 'selling_price', 'km_driven', 'fuel', 'seller_type',
       'transmission', 'owner', 'mileage', 'engine', 'max_power', 'torque',
       'seats'],
      dtype='object')

In [18]:
car_details_v4.sample(5)

Make                                    Model    Price  Year  \
1502  Mercedes-Benz  GLC 220d 4MATIC Progressive [2019-2021]  5200000  2019   
615   Maruti Suzuki                  Celerio ZXi [2017-2019]   445000  2018   
850        Mahindra                        Scorpio LX BS-III   365000  2009   
1645  Maruti Suzuki                      Wagon R LXI CNG (O)   380000  2018   
1860         Toyota       Innova 2.8 ZX AT 7 STR [2016-2020]  2150000  2017   

      Kilometer Fuel Type Transmission   Location   Color   Owner Seller Type  \
1502      10000    Diesel    Automatic      Delhi    Blue   First  Individual   
615       71000    Petrol       Manual   Ludhiana   White   First  Individual   
850       90000    Diesel       Manual    Lucknow   Green  Second  Individual   
1645      68320       CNG       Manual  Faridabad  Silver   First  Individual   
1860      45000    Diesel    Automatic     Mumbai    Grey   First  Individual   

       Engine           Max Power         Max Torque Drivetrain  Length  \
1502  1950 cc  192 bhp @ 3800 rpm  400 Nm @ 1600 rpm        AWD  4658.0   
615    998 cc   67 bhp @ 6000 rpm   90 Nm @ 3500 rpm        FWD  3600.0   
850   2179 cc  120 bhp @ 4000 rpm  290 Nm @ 1800 rpm        RWD  4430.0   
1645   998 cc   58 bhp @ 6200 rpm   77 Nm @ 3500 rpm        FWD  3599.0   
1860  2755 cc  172 bhp @ 3400 rpm  360 Nm @ 1200 rpm        RWD  4735.0   

       Width  Height  Seating Capacity  Fuel Tank Capacity  
1502  1890.0  1644.0               5.0                66.0  
615   1600.0  1560.0               5.0                35.0  
850   1817.0  1975.0               8.0                60.0  
1645  1495.0  1700.0               5.0                35.0  
1860  1830.0  1795.0               7.0                55.0

In [19]:
car_details_v4.describe(include='all').T

count unique                 top  freq            mean  \
Make                  2059     33       Maruti Suzuki   440             NaN   
Model                 2059   1050  X1 sDrive20d xLine    15             NaN   
Price               2059.0    NaN                 NaN   NaN  1702991.696455   
Year                2059.0    NaN                 NaN   NaN     2016.425449   
Kilometer           2059.0    NaN                 NaN   NaN    54224.714424   
Fuel Type             2059      9              Diesel  1049             NaN   
Transmission          2059      2              Manual  1133             NaN   
Location              2059     77              Mumbai   342             NaN   
Color                 2059     17               White   802             NaN   
Owner                 2059      6               First  1619             NaN   
Seller Type           2059      3          Individual  1997             NaN   
Engine                1979    108             1197 cc   231             NaN   
Max Power             1979    335   89 bhp @ 4000 rpm    90             NaN   
Max Torque            1979    290   200 Nm @ 1750 rpm    90             NaN   
Drivetrain            1923      3                 FWD  1330             NaN   
Length              1995.0    NaN                 NaN   NaN     4280.860652   
Width               1995.0    NaN                 NaN   NaN      1767.99198   
Height              1995.0    NaN                 NaN   NaN     1591.735338   
Seating Capacity    1995.0    NaN                 NaN   NaN        5.306266   
Fuel Tank Capacity  1946.0    NaN                 NaN   NaN        52.00221   

                               std      min       25%       50%        75%  \
Make                           NaN      NaN       NaN       NaN        NaN   
Model                          NaN      NaN       NaN       NaN        NaN   
Price               2419880.635434  49000.0  484999.0  825000.0  1925000.0   
Year                      3.363564   1988.0    2014.0    2017.0     2019.0   
Kilometer             57361.721314      0.0   29000.0   50000.0    72000.0   
Fuel Type                      NaN      NaN       NaN       NaN        NaN   
Transmission                   NaN      NaN       NaN       NaN        NaN   
Location                       NaN      NaN       NaN       NaN        NaN   
Color                          NaN      NaN       NaN       NaN        NaN   
Owner                          NaN      NaN       NaN       NaN        NaN   
Seller Type                    NaN      NaN       NaN       NaN        NaN   
Engine                         NaN      NaN       NaN       NaN        NaN   
Max Power                      NaN      NaN       NaN       NaN        NaN   
Max Torque                     NaN      NaN       NaN       NaN        NaN   
Drivetrain                     NaN      NaN       NaN       NaN        NaN   
Length                  442.458507   3099.0    3985.0    4370.0     4629.0   
Width                   135.265825   1475.0    1695.0    1770.0     1831.5   
Height                  136.073956   1165.0    1485.0    1545.0     1675.0   
Seating Capacity           0.82217      2.0       5.0       5.0        5.0   
Fuel Tank Capacity       15.110198     15.0     41.25      50.0       60.0   

                           max  
Make                       NaN  
Model                      NaN  
Price               35000000.0  
Year                    2022.0  
Kilometer            2000000.0  
Fuel Type                  NaN  
Transmission               NaN  
Location                   NaN  
Color                      NaN  
Owner                      NaN  
Seller Type                NaN  
Engine                     NaN  
Max Power                  NaN  
Max Torque                 NaN  
Drivetrain                 NaN  
Length                  5569.0  
Width                   2220.0  
Height                  1995.0  
Seating Capacity           8.0  
Fuel Tank Capacity       105.0

In [23]:
car_details_v4.columns

Index(['Make', 'Model', 'Price', 'Year', 'Kilometer', 'Fuel Type',
       'Transmission', 'Location', 'Color', 'Owner', 'Seller Type', 'Engine',
       'Max Power', 'Max Torque', 'Drivetrain', 'Length', 'Width', 'Height',
       'Seating Capacity', 'Fuel Tank Capacity'],
      dtype='object')

| **Car data**      | **CAR DETAILS FROM CAR DEKHO** | **Car details v3** | **Car details v4** |
|-------------------|--------------------------------|--------------------|--------------------|
| 301 observaciones | 4340 observaciones             | 8128 observaciones | 2059 observaciones |
| Car_Name          | name                           | name               | Make + Model       |
| Year              | year                           | year               | Year               |
| Selling_Price     | selling_Price                  | selling_price      | Price              |
| Present Price     | -                              | -                  | -                  |
| Kms_Driven        | km_driven                      | km_driven          | Kilometer          |
| Fuel_Type         | fuel                           | fuel               | Fuel Type          |
| Seller_Type       | seller_type                    | seller_type        | Seller Type        |
| Transmission      | transmission                   | transmission       | Transmission       |
| Owner             | owner                          | owner              | Owner              |
| -                 | -                              | mileage            | -                  |
| -                 | -                              | engine             | Engine             |
| -                 | -                              | max_power          | Max Power          |
| -                 | -                              | torque             | Max Torque         |
| -                 | -                              | seats              | Seating Capacity   |
| -                 | -                              | -                  | Drivetrain         |
| -                 | -                              | -                  | Length             |
| -                 | -                              | -                  | Width              |
| -                 | -                              | -                  | Height             |
| -                 | -                              | -                  | Location           |
| -                 | -                              | -                  | Color              |
| -                 | -                              | -                  | Fuel Tank Capacity |



Luego de revisar superficialmente los datasets, decidimos empezar a trabajar con `Car details v3`. El motivo es que cuenta con la mayor cantidad de observaciones (8128), y además suma 5 features con respecto a `car data` y `CAR DETAILS FROM CAR DEKHO`, que nos parece pueden resultar reelevantes en el entrenamiento de un modelo.

`Car details v4`, aunque con solo 2059 observaciones, cuenta con otras 7 features, algunas que pueden asumirse como interesantes para calcular el precio. Además, este dataset cuenta con valores de autos más actualizados que el dataset v3, por lo tanto podría suponerse que genere un mejor modelo para predecir el valor de autos de los últimos años. Eventualmente vamos a intentar explorar un poco las posibilidades con este dataset también, aunque manteniendo mayor foco en el dataset v3.